### PR TITLE
feat: printer calibration tool

### PR DESCRIPTION
**Pull request by andrushechka-b**
The work is still in progress. Please review and check my code if I go in right direction.
There is also a service tool that was used to adjust the calibration process. Later I suppose it will be removed for the standard user.

### CODE REVIEW

File: src/calibration/controller.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        self.view.btnNext.setEnabled(False)
+        self.view.progressBar.setVisible(True)
+
+    def clickFinish(self):
+        self.steps[self.step].printerMethod()
+        self.view.close()
+
+    def closeEvent(self, event):
+        self.worker_thread.terminate()
+        self.worker_thread.wait()
```
Comment (l1va) [inline src/calibration/controller.py:104]: you can put everything inside the worker I think

File: src/calibration/reprapfirmware_lsq.py [hunk 2]
```diff
@@ -0,0 +1,555 @@
...
+
+__author__ = 'Pedro'
+"""
+Derived from https://github.com/payala/DeltaTuner by @payala
+
+Delta printer least-squares calibration calculator
+
+This calculator implements the least-squares delta calibration algorithm that RepRapFirmware has built-in.
+I have provided it as a service for those not running RepRapFirmware.
+You can choose to calibrate the following parameters:
```
Comment (l1va) [inline src/calibration/reprapfirmware_lsq.py:12]: can be fixed

File: src/calibration/steps.py [hunk 3]
```diff
@@ -0,0 +1,405 @@
...
+
+    def __len__(self):
+        return len(self._steps)
+
+
+class Step:
+    def __init__(self, num, stepType='text', parent=None):
+        self.num = num
+        self.type = stepType
+        self.setLang('ru')
```
Comment (l1va) [inline src/calibration/steps.py:41]: get lang from the settings?

File: src/calibration/steps.py [hunk 4]
```diff
@@ -0,0 +1,405 @@
...
+            num_factors=7,
+        )
+        tuner.set_firmware("RRF")
+
+        points = tuner.get_probe_points()
+
+        # collect points for delta calibration
+        res = self.printer.defDelta(posZ=self.container.initZ, points=points)
+        for p in res:
+            print(p)
```
Comment (l1va) [inline src/calibration/steps.py:169]: remove me?

File: src/calibration/view.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        mainLayout.addLayout(btnLayout)
+
+        self.label = label
+        # self.btnBack = btnBack
+        self.btnNext = btnNext
+        self.btnCancel = btnCancel
+        self.btnFinish = btnFinish
+        self.progressBar = progressBar
+
+        self.setLang('ru')
```
Comment (l1va) [inline src/calibration/view.py:53]: hardcoded lang again

File: src/window.py [hunk 6]
```diff
@@ -131,6 +131,10 @@ def __init__(self, parent=None):
         self.slicing_info_action = QAction(self.locale.SlicerInfo, self)
         file_menu.addAction(self.slicing_info_action)
 
+        tools_menu = bar.addMenu(self.locale.Tools)
+        self.calibration_action = QAction(self.locale.Calibration, self)
+        tools_menu.addAction(self.calibration_action)
```
Comment (l1va) [inline src/window.py:136]: only calibration, should we add service too?

**Review by l1va — APPROVED** [review]
overall can be merged but consider the comments

File: src/calibration/reprapfirmware_lsq.py (hunk 2 above)
Comment (andrushechka-b) [reply src/calibration/reprapfirmware_lsq.py:12]: This is original author docstring. What if we just remove the last sentence?

File: src/calibration/view.py [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+    def setLang(self, lang):
+        if lang == 'ru':
```
Comment (lvjonok) [inline src/calibration/view.py:56]: why not to use locales?

File: src/printer/printer.py [hunk 8]
```diff
@@ -0,0 +1,1547 @@
...
+            ( 35, -57, -24), # noqa
+            ( 35,  42,  20), # noqa
+        )
+
+        points3 = (
+            (-35, -5, -35), # noqa
+            (-35, 65,  50), # noqa
+            ( 35, -5, -35), # noqa
+            ( 35, 65,  50), # noqa
+        )"""
```
Comment (lvjonok) [inline src/printer/printer.py:491]: just one big comment?)

File: src/printer/printer.py [hunk 9]
```diff
@@ -0,0 +1,1547 @@
...
+                X, Y, Z = point
+                moveTo(X, Y, Z, F=2000)
+
+            moveTo(Z=100, F=2000)
+            moveTo(X=0, Y=0)
+            tiltBed(V=0)
+
+        return
+
+        points = (
```
Comment (lvjonok) [inline src/printer/printer.py:1542]: seems it will not execute

**Review by lvjonok — APPROVED** [review]
I think we should merge and leave it

**Pull request merged**